drivers/dma/Kconfig.stm32

```diff
@@ -23,9 +23,10 @@ config DMA_STM32_V1
 
 config DMA_STM32_V2
 	bool
-	depends on SOC_SERIES_STM32F0X || SOC_SERIES_STM32F1X || SOC_SERIES_STM32F3X || SOC_SERIES_STM32L0X || SOC_SERIES_STM32L4X
```

Comment:
Ideally, this patch should be identical to commit from https://github.com/zephyrproject-rtos/zephyr/pull/20824
and this PR should be based on top and advertised as such.

Reply:
done, see commit 1423d4ed08abdff9125689d1ff2b48b3a1c38e9f